### ISSUE TITLE
Create "meanF" function

### ISSUE THREAD

**Issue by jonesor** [jonesor/Rcompadre#6]
Function to produce mean fecundity matrices from individual matrices.

**Comment by patrickbarks** [comment]
@DannyLBuss I started editing this function, and perhaps got carried away. I hope it's a bit simpler now, but let me know what you think.

I changed the functionality to return a mean of mean matFs instead of a random matF, and I also changed it to return a meanMatF for every single row of the database that gets passed, rather than for only a subset of "Individual" matrices. I figure the user can subset to "Individual" at a later point, if they so wish. Let me know what you think, and/or if you prefer the original function.

(Also, I changed the name to appendMeanMatF)

**Comment by patrickbarks** [comment]
I propose dropping the `meanMatF` function in favour of a more general function that takes a list of matrices (of equal dimension) and returns the element-wise mean matrix (e.g. `averageMats`).

The problem I see with `meanMatF` is that there are many different grouping levels at which a user might want to calculate a mean matrix, depending on the analysis, e.g.

- `SpeciesAuthor` x `MatrixPopulation`
- `SpeciesAuthor` x `MatrixPopulation` x `MatrixTreatment`
- `SpeciesAuthor` x `MatrixStartYear`
- `SpeciesAuthor` x `MatrixStartYear` x `<user-derived column>`

There's also the complication of how to deal with matrix dimension mismatch, if it pops up at the desired level. I think making `meanMatF` sufficiently flexible would also make it overly complicated.

With the tidy version of db (#27) and a function like `averageMats`, it would be easier for a user to average any type of matrix at any given group level with their own 'split-apply-combine' operation, e.g.
```
db %>% 
  group_by(SpeciesAuthor, MatrixDimension) %>% 
  mutate(matF_mean = averageMats(matF)) %>% 
  ungroup()
```
I suggest we replace `meanMatF` with `averageMats`, and add an example like this to the vignette (or website, etc.).

(Side note: `popbio` has a `mean.list` S3 method that does the same thing as `averageMats`, but I think it would be preferable to have a separate standalone function).

**Comment by patrickbarks** [comment]
`cdb_mean_matF` and `mpm_mean` can be used to produce mean fecundity matrices. My issues from the previous comment are resolved by #48, so closing this.
